### PR TITLE
Simple Security manager for UDFS

### PR DESCRIPTION
**Pull request by dguy**
### Description 
Add a security manager that blocks UDFs from calling `System.exit(..)` and from forking processes

### Testing done 
Added tests for the security manager and ensure everything else still works

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by dguy** [comment]
@hjafarpour @apurvam after the discussion yesterday i decided to dial this back to just check for `System.exit()` and process execution.

File: ksql-engine/src/main/java/io/confluent/ksql/security/ExtensionSecurityManager.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import java.security.PermissionCollection;
+import java.security.Permissions;
+import java.security.Policy;
+import java.security.ProtectionDomain;
+import java.util.Stack;
+
+import io.confluent.ksql.function.udf.PluggableUdf;
+
+/**
+ * A simple security manger extension to block UDFs from calling
```
Comment (apurvam) [inline ksql-engine/src/main/java/io/confluent/ksql/security/ExtensionSecurityManager.java]: typo: manager

File: ksql-engine/src/main/java/io/confluent/ksql/security/ExtensionSecurityManager.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  public synchronized void pushInUdf() {
+    if (callerIsAllowed()) {
+      if (UDF_IS_EXECUTING.get() == null) {
+        UDF_IS_EXECUTING.set(new Stack<>());
+      }
+      UDF_IS_EXECUTING.get().push(true);
+    }
+  }
+
+  public void popInUdf() {
```
Comment (apurvam) [inline ksql-engine/src/main/java/io/confluent/ksql/security/ExtensionSecurityManager.java]: maybe rename this to `popOutUdf`. Seems closer to what is actually happening.

File: ksql-engine/src/main/java/io/confluent/ksql/security/ExtensionSecurityManager.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    if (callerIsAllowed()) {
+      final Stack<Boolean> stack = UDF_IS_EXECUTING.get();
+      if (stack != null && !stack.isEmpty()) {
+        UDF_IS_EXECUTING.get().pop();
```
Comment (apurvam) [inline ksql-engine/src/main/java/io/confluent/ksql/security/ExtensionSecurityManager.java]: use the stack instance from above?

File: ksql-engine/src/main/java/io/confluent/ksql/security/ExtensionSecurityManager.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    final Stack<Boolean> executing = UDF_IS_EXECUTING.get();
+    return executing != null && !executing.isEmpty();
+  }
+
+  /**
+   * Check if the caller is a PluggableUdf. It will be the third
+   * item in the class array.
+   * @return true if caller is allowed
+   */
+  private boolean callerIsAllowed() {
```
Comment (apurvam) [inline ksql-engine/src/main/java/io/confluent/ksql/security/ExtensionSecurityManager.java]: I think the intent of this method is to support enable security policy enforcemnet only for pluggable udfs. If so, maybe this should be renamed to `shouldValidateCaller` or something similar?

File: ksql-engine/src/test/java/io/confluent/ksql/function/udf/PluggableUdfTest.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    System.setSecurityManager(ExtensionSecurityManager.INSTANCE);
+  }
+
+  @After
+  public void after() {
+    System.setSecurityManager(securityManager);
+  }
+
+  @Test(expected = SecurityException.class)
+  public void shouldEnableUdfSecurity() {
```
Comment (apurvam) [inline ksql-engine/src/test/java/io/confluent/ksql/function/udf/PluggableUdfTest.java]: Hmm. This seems identical to the test case in `ExtensionSecurityManagerTest`. Any particular reason to duplicate?

Comment (dguy) [reply ksql-engine/src/test/java/io/confluent/ksql/function/udf/PluggableUdfTest.java]: nope it is just where i first wrote it. Can remove

**Pull request merged**